src/components/Providers/CourseSearchingProvider.js

```diff
@@ -13,7 +13,7 @@ const CourseSearchingContext = createContext({
     show_selected_courses: false,
     only_show_not_conflicted_courses: false,
     sync_add_to_nol: false,
-    strict_search_mode: false,
+    strict_search_mode: true,
```

Comment:
There's no effect after this change, if you want to set default value to `true`. Modify line59 instead